Docker/dev.Dockerfile

```diff
@@ -10,10 +10,9 @@
 #tag sdk:8.0-alpine
 FROM mcr.microsoft.com/dotnet/sdk:8.0.203-alpine3.18@sha256:2a8dca3af111071172b1629c12eefaeca0d6c2954887c4489195771c9e90833c as buildBase
 
-FROM buildbase as build
 RUN apk --no-cache add curl=~8
 
-FROM build AS publish
+FROM buildbase AS publish
```

Comment:
Hadolint is still flagging this; [it shouldn't](https://github.com/hadolint/hadolint/wiki/DL3006). Perhaps it is confused because _copy_ from `publish` rather than build another layer from it? Anyway, now we will have to silence Hadolint on this issue. I've added a commit that takes care of it.

I also added a commit that removes unnecessary comments.